src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringContainsCharOverloadWithSingleCharactersAnalyzer.cs

```diff
@@ -0,0 +1,103 @@
+        private static readonly LocalizableString s_localizableDescription_CA1846 = new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseStringContainsCharOverloadWithSingleCharactersDescription), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor s_rule_CA1846 = DiagnosticDescriptorHelper.Create(
+            CA1846,
+            s_localizableTitle_CA1846,
+            s_localizableMessage_CA1846,
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            description: s_localizableDescription_CA1846,
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(s_rule_CA1846);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterCompilationStartAction(CheckIfRuleIsApplicableAndRegister);
+        }
+
+        private void CheckIfRuleIsApplicableAndRegister(CompilationStartAnalysisContext context)
+        {
+            var stringType = context.Compilation.GetSpecialType(SpecialType.System_String);
+            var charType = context.Compilation.GetSpecialType(SpecialType.System_Char);
+
+            // Bail out of further evaluations in special cases where string/char types arent even available
+            if (stringType is null || charType is null)
+                return;
+
+            var ruleIsApplicable = stringType.GetMembers("Contains").OfType<IMethodSymbol>().Any(m =>
+            {
+                if (m.Parameters.Length > 0)
+                {
+                    return m.Parameters[0].Type.SpecialType == SpecialType.System_Char;
+                }
+                return false;
+            });
+
+            if (ruleIsApplicable)
+                context.RegisterOperationAction(AnalyseContainsUsage, OperationKind.Invocation);
+        }
+
+        private void AnalyseContainsUsage(OperationAnalysisContext context)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            var argumentOperation = invocationOperation.Arguments.GetArgumentForParameterAtIndex(0);
+            if (ShouldAnalyzeInvocations(invocationOperation) && CheckForViolationAndTransformLiteral(argumentOperation, out var sourceString))
+            {
+                var propertyBuilder = ImmutableDictionary.CreateBuilder<string, string?>(StringComparer.Ordinal);
+                propertyBuilder.Add(SourceStringPropertyKey, sourceString);
+                if (IsNamedArgument(argumentOperation, out var name))
+                    propertyBuilder.Add(ArgumentNamePropertyKey, name);
```

Comment:
Seems very weird to have the `IsNamedArgument` check in the analyzer, as the analyzer does not need this information at all. Can you instead move this code to the fixer and also remove `ArgumentNamePropertyKey`?

Reply:
Sounds good, thanks!